### ISSUE TITLE
Numbered op list

### ISSUE THREAD

**Issue by wayfu** [vsariola/sointu#115]
To make it easier to keep track of multiple same ops in a row, it would be good to have a numbered list of ops in the instrument!

**Comment by wayfu** [comment]
another cool feature would be the option to assign a color to ops in the op list

**Comment by vsariola** [comment]
> To make it easier to keep track of multiple same ops in a row, it would be good to have a numbered list of ops in the instrument!

You mean the instrument editor should show the index of the unit to it left? That's super easy to do, but I wasn't sure that alone is helping too much + adds visual clutter in the GUI (it starts to be pretty busy already!).

> another cool feature would be the option to assign a color to ops in the op list

I like this a lot more. One could color units to remind which one is which. I'll probably look into this.

**Comment by wayfu** [comment]
> > To make it easier to keep track of multiple same ops in a row, it would be good to have a numbered list of ops in the instrument!
> 
> You mean the instrument editor should show the index of the unit to it left? That's super easy to do, but I wasn't sure that alone is helping too much + adds visual clutter in the GUI (it starts to be pretty busy already!).

maybe make it an option then? it's how we choose targets in the send op, for example, and it would be good to have a direct reference in mind instead of say, counting filters to find the fifth one :)

**Comment by vsariola** [comment]
Question: now that there is the ability to add a comment to each unit, is that enough? Color seems a bit redundant, considering that a comment can be used for any text. Similarly, showing the line number seems not that useful as the line numbers tend to change all the time as you add units and delete units, so in practice finding the right send target based on line number seems not that easy.

But if e.g. you have main envelope and another envelope for pitch drop, you could now comment them as "main" and "pitchdrop" and then finding them will be easy always.

I think this can be closed?

**Comment by wayfu** [comment]
i can't find the comment-per-unit feature, is that in the release? 

slot numbers are good because they happen automatically and don't require people to manually add comments to individual units; additionally, the send dropdown already shows them - nothing could be more intuitive, i think :)

**Comment by vsariola** [comment]
No, it's only in the latest master still, try the binaries from here: https://github.com/vsariola/sointu/actions/runs/11340616285

**Comment by wayfu** [comment]
ah, got it working, thank you. i still think it makes sense to add the slot numbers, but this is already very nice. one detail that i think might be necessary is to make the unit pane resizable in width, there's not much space for unit comments: 
![image](https://github.com/user-attachments/assets/351b52ea-f841-4835-9772-a0e425ef4af2)

**Comment by LeStahL** [comment]
The comment length seems to be capped at 16 chars. I love the feature tho, super cool that this works now :)

**Comment by vsariola** [comment]
The cap was kinda planned because my original plan was to show very little of the comment, just to avoid making the unit list resizeable so the user understands that the comments are mainly for distinguishing units, not for documentation (that is better done in the instrument comments). I can remove the cap and stop the hand holding & just let the user do whatever she wants.

**Comment by wayfu** [comment]
maybe there'll be enough space to display the slot number then?
```
  🥺 
 👉👈
```

**Comment by vsariola** [comment]
Now with the new GUI with the signal lines showing which send connects to which target port, and also the incoming / outgoing global modulations being shown with text where they go and where they come from, I am considering this unit numbering unnecessary. As I understood, the unit numbering would have been useful mainly to help picking right send targets.

However, I still consider the unit coloring a valid enhancement, as this might help the user to remember the differences between two similar units. So I am keeping this issue open, but just for the coloring part.
